### PR TITLE
report: add markdown detail type

### PR DESCRIPTION
**Pull request by paulirish**
@adamraine's csp-xss audit has lots of strings that would benefit from monospace formatting.

here's a before and after: (updated a few times)

image

I was thinking about if all `type: text` should get this upgrade.. and it's compelling.. but i think a dedicated `type: markdowntext` is slightly better, especially for any downstream consumers of our JSON.

### CODE REVIEW

File: lighthouse-core/report/html/renderer/details-renderer.js [hunk 1]
```diff
@@ -287,6 +299,9 @@ class DetailsRenderer {
         const strValue = String(value);
         return this._renderText(strValue);
       }
+      case 'markdowntext': {
```
Comment (connorjclark) [inline lighthouse-core/report/html/renderer/details-renderer.js:302]: `markdown` seems pretty unambiguous to me wdyt

**Review by connorjclark — CHANGES_REQUESTED** [review]
lgtm but i think background color would be dope

File: lighthouse-core/report/html/renderer/details-renderer.js (hunk 1 above)
Comment (paulirish) [reply lighthouse-core/report/html/renderer/details-renderer.js]: done

**Comment by paulirish** [comment]
> lgtm but i think background color would be dope


bg color sg. done

**Review by brendankenny — COMMENTED** [review]
needs tests :)

> I was thinking about if all type: text should get this upgrade.. and it's compelling.. but i think a dedicated type: markdowntext is slightly better, especially for any downstream consumers of our JSON.

My vote is add formatting to `'text'` 😬 

I can see your reasoning, but I also can't think of a case when we'd ever want to use a plain text table entry without inline code support after this (except "it doesn't need inline code yet"), so it does mean we'll have this legacy split forever :/

Downstream people rarely look at the heading anyways, just the json path, so it seems unlikely to hurt anyone peering into existing items. Also, at worst, they get some extra backticks in their strings.

OTOH downstream users _will_ have to rev the renderer for sites like web.dev or they'll hit the dreaded `[renderUnknown()](https://github.com/GoogleChrome/lighthouse/blob/aae0d0e984c2cec46a796b1d38b779b06f9438ba/lighthouse-core/report/html/renderer/details-renderer.js#L211-L221)` for `markdown`.

File: lighthouse-core/audits/csp-xss.js [hunk 2]
```diff
@@ -143,7 +143,7 @@ class CspXss extends Audit {
     /** @type {LH.Audit.Details.Table['headings']} */
```
Comment (brendankenny) [inline lighthouse-core/audits/csp-xss.js:143]: should also backtick the `<meta>` in this file

File: lighthouse-core/lib/csp-evaluator.js [hunk 3]
```diff
@@ -21,32 +21,32 @@ const i18n = require('../lib/i18n/i18n.js');
...
+  unsafeInline: '`unsafe-inline` allows the execution of unsafe in-page scripts ' +
     'and event handlers. Consider using CSP nonces or hashes to allow scripts individually.',
   /** Message shown when a CSP is not backwards compatible with browsers that do not support CSP nonces/hashes. Shown in a table with a list of other CSP vulnerabilities and suggestions. "CSP" stands for "Content Security Policy". "'unsafe-inline'", "nonces", and "hashes" do not need to be translated. */
-  unsafeInlineFallback: 'Consider adding \'unsafe-inline\' (ignored by browsers supporting ' +
+  unsafeInlineFallback: 'Consider adding `unsafe-inline` (ignored by browsers supporting ' +
     'nonces/hashes) to be backward compatible with older browsers.',
   /** Message shown when a CSP is not backwards compatible with browsers that do not support the 'strict-dynamic' keyword. Shown in a table with a list of other CSP vulnerabilities and suggestions. "CSP" stands for "Content Security Policy". "http:", "https:", and "'strict-dynamic'" do not need to be translated. */
   allowlistFallback: 'Consider adding https: and http: URL schemes (ignored by browsers ' +
```
Comment (adamraine) [inline lighthouse-core/lib/csp-evaluator.js:44]: http and https?

File: lighthouse-core/lib/csp-evaluator.js [hunk 4]
```diff
@@ -21,32 +21,32 @@ const i18n = require('../lib/i18n/i18n.js');
...
+    'Consider setting `base-uri` to `none` or `self`.',
   /** Message shown when a CSP does not have a script-src directive. Shown in a table with a list of other CSP vulnerabilities and suggestions. "CSP" stands for "Content Security Policy". "script-src" does not need to be translated. */
-  missingScriptSrc: 'script-src directive is missing. ' +
+  missingScriptSrc: '`script-src` directive is missing. ' +
     'This can allow the execution of unsafe scripts.',
   /** Message shown when a CSP does not have a script-src directive. Shown in a table with a list of other CSP vulnerabilities and suggestions. "CSP" stands for "Content Security Policy". "object-src" and "'none'" do not need to be translated. */
-  missingObjectSrc: 'Elements controlled by object-src are considered legacy features. ' +
-    'Consider setting object-src to \'none\' to prevent the injection of ' +
+  missingObjectSrc: 'Elements controlled by `object-src` are considered legacy features. ' +
+    'Consider setting `object-src` to `none` to prevent the injection of ' +
```
Comment (adamraine) [inline lighthouse-core/lib/csp-evaluator.js:32]: I think the extra single quotes are are necessary. `'none'` is a keyword whereas `none` is a syntax error.

**Comment by paulirish** [comment]
i haven't cared much about this since march and it wasn't a slam dunk, so i'm gonna close without expectation of resuming it.

**Pull request closed without merging**